contrib/core-contract-tests/tests/pox_4_test.clar

```diff
@@ -0,0 +1,123 @@
+)
+
+(define-public (test-invalid-pox-addr-hashbytes-length)
+	(let (
+		(test-lengths (list u0 u1 u2 u3 u4 u5 u6 u7 u8 u9 u10 u11 u12 u13 u14 u15 u16 u17 u18 u19 u20 u21 u22 u23 u24 u25 u26 u27 u28 u29 u30 u31 u32))
+		(length-20-valid (list
+			false false false false false false false false false false false false false false false false
+			false false false false true  false false false false false false false false false false false false
+			))
+		(length-32-valid (list
+			false false false false false false false false false false false false false false false false
+			false false false false false false false false false false false false false false false false true
+			))
+		(length-all-invalid (list
+			false false false false false false false false false false false false false false false false
+			false false false false false false false false false false false false false false false false false
+			))
+		)
+		(asserts! (is-eq (map check-pox-addr-hashbytes-iter test-lengths (buff-repeat 0x00 (len test-lengths))) length-20-valid)
+			(err "Only length 20 should be valid for version 0x00")
+		)
+		(asserts! (is-eq (map check-pox-addr-hashbytes-iter test-lengths (buff-repeat 0x01 (len test-lengths))) length-20-valid)
+			(err "Only length 20 should be valid for version 0x01")
+		)
+		(asserts! (is-eq (map check-pox-addr-hashbytes-iter test-lengths (buff-repeat 0x02 (len test-lengths))) length-20-valid)
+			(err "Only length 20 should be valid for version 0x02")
+		)
+		(asserts! (is-eq (map check-pox-addr-hashbytes-iter test-lengths (buff-repeat 0x03 (len test-lengths))) length-20-valid)
+			(err "Only length 20 should be valid for version 0x03")
+		)
+		(asserts! (is-eq (map check-pox-addr-hashbytes-iter test-lengths (buff-repeat 0x04 (len test-lengths))) length-20-valid)
+			(err "Only length 20 should be valid for version 0x04")
+		)
+		(asserts! (is-eq (map check-pox-addr-hashbytes-iter test-lengths (buff-repeat 0x05 (len test-lengths))) length-32-valid)
+			(err "Only length 20 should be valid for version 0x05")
+		)
+		(asserts! (is-eq (map check-pox-addr-hashbytes-iter test-lengths (buff-repeat 0x06 (len test-lengths))) length-32-valid)
+			(err "Only length 20 should be valid for version 0x06")
+		)
+		(asserts! (is-eq (map check-pox-addr-hashbytes-iter test-lengths (buff-repeat 0x07 (len test-lengths))) length-all-invalid)
+			(err "No length should be valid for version 0x07")
+		)
+		(ok true)
+	)
+)
+
+(define-public (test-invalid-lock-height-too-low)
+	(let
+		((actual (contract-call? .pox-4 check-pox-lock-period u0)))
+		(asserts! (not actual) (err u111))
+		(ok true)))
+
+(define-public (test-invalid-lock-height-too-high)
+	(let
+		((actual (contract-call? .pox-4 check-pox-lock-period u13)))
+		(asserts! (not actual) (err u111))
+		(ok true)))
+
+(define-public (test-get-total-ustx-stacked)
+	(begin 
+		;; @continue
```

Comment:
What does this do? I don't see it documented in clarunit.

Reply:
Some of this code is leftover code perhaps. I did not work on this portion of clarunit so I'm summoning @friedger. Do we still need this?

Reply:
Should be removed